src/cmd/ksh93/sh/string.c

```diff
@@ -335,7 +335,8 @@ char *sh_fmtstr(const char *string, int quote) {
         while (op = cp, c = mbchar(cp)) {
             state = 1;
             switch (c) {
-                case ('a'==97?'\033':39): {
```

Comment:
This is testing whether we're on a machine that uses ASCII or EBCDIC. If the latter `a` has value 129 and escape is 39. Note, however, that Linux on IBM zSeries machines uses ASCII so this compatibility code isn't needed in that environment. It's only needed if using ksh on something like z/OS UNIX. Personally I'd like to remove the EBCDIC compatibility code but it's probably something we should discuss in an issue dedicated to the question.

Reply:
Concur on removing EBCDIC support, would be interested to hear what others think.